SharpPcap/LibPcap/LibPcapLiveDevice.cs

```diff
@@ -204,7 +134,7 @@ public override void Open(DeviceMode mode, int read_timeout, MonitorMode monitor
         /// <param name="kernel_buffer_size">
         /// A <see cref="uint"/>
         /// </param>
-        public override void Open(DeviceMode mode, int read_timeout, MonitorMode monitor_mode, uint kernel_buffer_size)
+        public override void Open(OpenFlags flags = OpenFlags.None, int read_timeout = 1000, MonitorMode monitor_mode = MonitorMode.Inactive, uint kernel_buffer_size = 0)
```

Comment:
If `flags` contain a flag other than `Promiscuous` then `pcap_open` should be used,

Reply:
Good call on that.

Yeah, sucks for osx users with old ass versions of libpcap currently. Maybe they'll update at some point.

What to do for osx if someone passes in those flags? Call pcap_open() let the exception go upwards or what?

Reply:
You can use the logic I used to detect if `pcap_sendqueue_transmit` is available in `SendQueue.GetIsHardwareAccelerated`
https://github.com/chmorgan/sharppcap/blob/master/SharpPcap/LibPcap/SendQueue.cs#L36

Reply:
What would we do if the user is asking for a feature that we can't provide, like NoCaptureLocal? Throw an unsupported exception in that case?

Reply:
* We could use the name `DeviceModes` for the enum.
* We could add `DeviceModes Pcap.GetSupportedModes(PcapInterface, DeviceModes)` that performs os & libpcap version check and inform about the supported modes.
*  We simply pass the flags as is, and use the best available function, after that it's up to libpcap to use, ignore or reject the flag.

Reply:
@kayoub5 I like DeviceModes. Let me rename that and re-push so you can review. I'm ok with deferring the checking for pcap_open() until a later PR so we can get this in and get things moving.